### PR TITLE
 Fixing the problems and Applying the comments of commentors

### PR DESCRIPTION
**Pull request by SSTechInd-Tec**
I use the router revalidation in for On-Demand Revalidation. Use below URL for making a revalidation request based on tags 
http://localhost:3000/api/revalidate?tag=reload

### CODE REVIEW

File: app/api/data.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+export const data = [
```
Comment (hakima3) [inline app/api/data.ts]: move api folder from app folder

File: app/components/DeleteTodo.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+"use client"
+
+import { useRouter } from "next/navigation"
+
+export default function DeleteTodo(props:any) {
```
Comment (hakima3) [inline app/components/DeleteTodo.tsx]: Don't use any as type and also restructure the props object and read author and id 
``` typescript
export default function DeleteTodo({author , id} : {author: string, id : string})
```

File: app/components/UpdateTodo.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+"use client"
+
+import { useRouter } from "next/navigation"
+
+export default function UpdateTodo(props:any) {
+
+    const router = useRouter()
+
+    const handleClick = async (e:any) => {
```
Comment (hakima3) [inline app/components/UpdateTodo.tsx]: use a type instead of any

File: app/components/UpdateTodo.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+"use client"
+
+import { useRouter } from "next/navigation"
+
+export default function UpdateTodo(props:any) {
```
Comment (hakima3) [inline app/components/UpdateTodo.tsx]: the same as above use destructuring

File: app/components/addForm.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        })
+
+        if (res.ok) {
+            router.push("/todos/list/")
+        }
+
+        const res2 = await fetch("http://localhost:3000/api/")
+
+        const d = await res2.json()
+        console.log(d)
```
Comment (hakima3) [inline app/components/addForm.tsx]: If you have not used this part pls remove and also don't push console.log to GitHub

File: app/todos/list/[author]/[id]/page.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import DeleteTodo from "@/app/components/DeleteTodo"
+
+const fetchData = async () => {
+    const res = fetch(`http://localhost:3000/api/`)
+    const data = (await res).json()
+
+    return data
+}
+
+const ViewOneTodo = async ({params}:any) => {
```
Comment (hakima3) [inline app/todos/list/[author]/[id]/page.tsx]: destructure params and put a good name for your function
it should like 
```typescript

const ViewTodo = async () => { }


```

**Review by hakima3 — CHANGES_REQUESTED** [review]

File: app/api/data.ts (hunk 1 above)
Comment (SSTechInd-Tec) [reply app/api/data.ts]: I can't move the api folder. If I do that, I can't access the api folder

**Pull request closed without merging**